temporalio/client.py

```diff
@@ -1612,6 +1615,259 @@ async def terminate(
+        ...
+
+    # Overload for string-name start update
+    @overload
+    async def start_update(
+        self,
+        update: str,
+        arg: Any = temporalio.common._arg_unset,
+        *,
+        args: Sequence[Any] = [],
+        id: Optional[str] = None,
+        result_type: Optional[Type] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowUpdateHandle[Any]:
+        ...
+
+    async def start_update(
+        self,
+        update: Union[str, Callable],
+        arg: Any = temporalio.common._arg_unset,
+        *,
+        args: Sequence[Any] = [],
+        id: Optional[str] = None,
+        result_type: Optional[Type] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowUpdateHandle[Any]:
+        """Send an update request to the workflow and return a handle to it.
+
+        This will target the workflow with :py:attr:`run_id` if present. To use a
+        different run ID, create a new handle with via :py:meth:`Client.get_workflow_handle`.
+
+        .. warning::
+           This API is experimental
+
+        .. warning::
+            WorkflowHandles created as a result of :py:meth:`Client.start_workflow` will
+            send updates to the latest workflow with the same workflow ID even if it is
+            unrelated to the started workflow.
+
+        Args:
+            update: Update function or name on the workflow.
+            arg: Single argument to the update.
+            args: Multiple arguments to the update. Cannot be set if arg is.
+            id: ID of the update. If not set, the server will set a UUID as the ID.
+            result_type: For string updates, this can set the specific result
+                type hint to deserialize into.
+            rpc_metadata: Headers used on the RPC call. Keys here override
+                client-level RPC metadata keys.
+            rpc_timeout: Optional RPC deadline to set for the RPC call.
+
+        Raises:
+            RPCError: There was some issue sending the update to the workflow.
+        """
+        return await self._start_update(
+            update,
+            arg,
+            args=args,
+            id=id,
+            wait_for_stage=temporalio.api.enums.v1.UpdateWorkflowExecutionLifecycleStage.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ACCEPTED,
+            result_type=result_type,
+            rpc_metadata=rpc_metadata,
+            rpc_timeout=rpc_timeout,
+        )
+
+    async def _start_update(
+        self,
+        update: Union[str, Callable],
+        arg: Any = temporalio.common._arg_unset,
+        *,
+        args: Sequence[Any] = [],
+        id: Optional[str] = None,
+        wait_for_stage: temporalio.api.enums.v1.UpdateWorkflowExecutionLifecycleStage.ValueType = temporalio.api.enums.v1.UpdateWorkflowExecutionLifecycleStage.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ADMITTED,
```

Comment:
I understand why you defaulted this to `ADMITTED` (it's what's most natural when asking for a handle) and I know the API is not exposed to users, but `ADMITTED` is completely unimplemented server-side and there is no agreed plan for how it would be implemented yet, and no timeline for implementing it, so maybe it's a bit odd for it to be in the SDK codebase? Maybe it should be a required positional arg.